### PR TITLE
[WIP] Web UI

### PR DESCRIPTION
**Pull request by VicDeo**
REQUIREMENTS
- [ ]  the updater has to have it's own html layout, the integration with the core admin page will be a simple link to open the index within the updater,  no json integration or what so ever

TODO:
- [x] authorization/admin privileges check; Moved to https://github.com/owncloud/updater/issues/220 
- [x] pass a path to oc instance somehow (For now we agreed that it always will be `updater` subdirectory and the only legit place to search for OC files is one dir above the file `updater/index.php`)

### CODE REVIEW

File: src/Controller/IndexController.php [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+		} else {
+			$this->request = $request;
+		}
+
+		$this->command = $this->request->postParameter('command');
+	}
+
+	public function dispatch(){
+		if (is_null($this->command)){
+			$templates = new Engine('/home/deo/public_html/updater/src/Resources/views/');
```
Comment (DeepDiver1975) [inline src/Controller/IndexController.php]: needs fixing :see_no_evil:

**Comment by karlitschek** [comment]
can't we do it without an .htaccess file? This is fragile and won't work in a lot of cases

**Comment by VicDeo** [comment]
@karlitschek this one is pushed unintentionally and will be removed later.
But the top-level htaccess needs to be changed anyway because Web UI is not accessible currently due to an url rewrite from  `domain.tld/updater/index.php` to `domain.tld/index.php/apps/updater/`.

**Comment by karlitschek** [comment]
@VicDeo O.K.
I think we should call the updater web ui with relative urls only and don't rely on any routing or redirects or rewriting. All this stuff might not work.

**Comment by VicDeo** [comment]
@DeepDiver1975 hardcoded path is fixed. 
Do you have any objections on merging  this PR to keep going forward with small steps?

**Comment by DeepDiver1975** [comment]
> But the top-level htaccess needs to be changed anyway because Web UI is not accessible currently due to an url rewrite from domain.tld/updater/index.php to domain.tld/index.php/apps/updater/.


why is there a rewrite necesssary?

the new updater will be accessible using  domain.tld/updater/index.php

I miss something I guess

**Comment by DeepDiver1975** [comment]
there is something wrong in loading the assets when accessing the updater in a subfolder

bildschirmfoto von 2016-01-28 13-36-56

**Comment by DeepDiver1975** [comment]
We need to switch all occ command invocations to actual web requests to the ownCloud instance.
We cannot spawn php processes within a web request.

Sounds like we need a web executor for occ commands ...

@karlitschek @PVince81 @icewind1991 @VicDeo ideas? objections?

**Comment by VicDeo** [comment]
@DeepDiver1975 
>why is there a rewrite necesssary?
the new updater will be accessible using domain.tld/updater/index.php
I miss something I guess

We agreed (in Skype) that it will be an URL like this: 
`domain.tld/updater` or `domain.tld/some_path/updater`

This approach requires an additional rewrite condition here
 `   RewriteCond %{REQUEST_FILENAME} !/updater/`
https://github.com/owncloud/core/blob/master/.htaccess#L61

This is also the answer to the assets loading issue

**Comment by VicDeo** [comment]
@DeepDiver1975 
>We need to switch all occ command invocations to actual web requests to the ownCloud instance.
We cannot spawn php processes within a web request.
Sounds like we need a web executor for occ commands ...

In this case I'd stay away from Symfony Console Application for web UI and move all logic from commands to abstraction layer leaving only IO operations inplace.

**Comment by DeepDiver1975** [comment]
> In this case I'd stay away from Symfony Console Application for web UI and move all logic from commands to abstraction layer leaving only IO operations inplace.

I see no need for this since we only need two different implementation in the updater for the occ commands - one using php process invocation and one which is performing a web request to the owncloud instance

**Comment by VicDeo** [comment]
OCC command list:
upgrade
status
config:list
app:disable 
app:enable 
app:getpath

**Comment by karlitschek** [comment]
@VicDeo Can we also have commands for
- Check if an update is available
- Download a new owncloud release only
- Generate a backup of the current installation

**Comment by DeepDiver1975** [comment]
> Check if an update is available
> Download a new owncloud release only

these are commands which are implemented within the updater - nothing which we want to rely on owncloud core

**Comment by DeepDiver1975** [comment]
> Generate a backup of the current installation

same from my pov

**Comment by karlitschek** [comment]
ahh sorry. maybe i misunderstood. :-) I mean that we need this in the updater. not in the core of course.

**Comment by VicDeo** [comment]
@karlitschek yes, this is a list of commands that we use with core `occ` command.

**Pull request merged**